File: processor/src/main/java/com/github/snowdream/toybricks/processor/BaseAnnotatedClass.java
package com.github.snowdream.toybricks.processor;

import javax.lang.model.element.TypeElement;

/**
 * Created by snowdream on 17/2/12.
 */
public abstract class BaseAnnotatedClass {

    protected TypeElement annotatedClassElement;

    /**
     *
     */
    public abstract void checkValid(ProcessorManager processorManager) throws ProcessingException;
}

File: processor/src/main/kotlin/com/github/snowdream/toybricks/processor/BaseAnnotatedClass.kt
package com.github.snowdream.toybricks.processor

import javax.lang.model.element.TypeElement

/**
 * Created by snowdream on 17/2/12.
 */
abstract class BaseAnnotatedClass {

    protected lateinit var annotatedClassElement: TypeElement

    /**

     */
    @Throws(ProcessingException::class)
    abstract fun checkValid(processorManager: ProcessorManager)
}
